### PR TITLE
[NodeCollector] Apply NodeRepository::collect() on Name node

### PR DESCRIPTION
**Pull request by samsonasik**
ref https://github.com/rectorphp/rector/pull/5558#discussion_r577085820

### CODE REVIEW

**Comment by samsonasik** [comment]
@TomasVotruba update Repository::collect to collect Name seems cause many test errors, it seems the naming make duplicate register keys.

**Comment by samsonasik** [comment]
use direct `$this->nodeNameResolver->getName($node)` seems the solution.

**Comment by samsonasik** [comment]
@TomasVotruba it works 🎉

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by samsonasik** [comment]
rebased.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: packages/node-collector/src/NodeCollector/NodeRepository.php [hunk 1]
```diff
@@ -149,17 +155,7 @@ public function collect(Node $node): void
...
-                return;
-            }
-
-            if (! $arrayCallable->isExistingMethod()) {
-                return;
-            }
-
-            $this->arrayCallablesByTypeAndMethod[$arrayCallable->getClass()][$arrayCallable->getMethod()][] = $arrayCallable;
-
+            $this->collectArray($node);
```
Comment (TomasVotruba) [inline packages/node-collector/src/NodeCollector/NodeRepository.php:158]: :+1:

**Comment by TomasVotruba** [comment]
Thank you Abdul :+1:

**Pull request merged**